libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/FinishBuildListener.kt

```diff
@@ -18,64 +18,89 @@ package org.jetbrains.kotlin.gradle.plugin
 
 import org.gradle.BuildAdapter
 import org.gradle.BuildResult
+import org.gradle.api.invocation.Gradle
 import org.gradle.api.logging.Logging
 import java.lang.ref.Reference
 import java.util.concurrent.ScheduledExecutorService
+import kotlin.text.MatchGroup
 
-class FinishBuildListener(var pluginClassLoader: ParentLastURLClassLoader?) : BuildAdapter() {
+internal fun getUsedMemoryKb(): Long {
+    System.gc()
+    val rt = Runtime.getRuntime()
+    return (rt.totalMemory() - rt.freeMemory()) / 1024
+}
+
+
+private fun comparableVersionStr(version: String) =
+        "(\\d+)\\.(\\d+).*"
+                .toRegex()
+                .match(version)
+                ?.groups
+                ?.drop(1)?.take(2)
+                // checking if two subexpression groups are found and length of each is >0 and <4
+                ?.let { if (it.all { (it?.value?.length() ?: 0).let { it > 0 && it < 4 }}) it else null }
+                ?.map { "%3s".format(it?.value ?: 0).replace(' ', '0') }
+                ?.joinToString(".")
```

Comment:
filterNotNull theoretically is not suitable, because it may drop groups in the middle, while I want the result be null if any of the group is null or have incorrect length.
Thanks for the padStart!

The expression would be probably simpler with toInteger conversion, but I assume it to be slow and less safe.